From 6eef48b3175a327623b44e6b7e39b8e5b7fe5644 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 24 Oct 2024 20:53:23 +0800
Subject: [PATCH 172/301] AOSCOS: drm: amdgpu: disable ABM (Adaptive Backlight
 Management) by default

Per report from a contributor, since v6.9, ABM was set to be automatically
enabled/disabled. It may have seemed like a good idea for brighter
screens, but the brightness adjustment is quite abrupt, which is quite
bothersome.

Disable this feature by default for now.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Link: https://t.me/c/1109254909/594612
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
index 9dbab53d48dc..8401b7b98422 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -883,9 +883,9 @@ module_param_named(visualconfirm, amdgpu_dc_visual_confirm, uint, 0444);
  * Defaults to -1, or auto. Userspace can only override this level after
  * boot if it's set to auto.
  */
-int amdgpu_dm_abm_level = -1;
+int amdgpu_dm_abm_level = 0;
 MODULE_PARM_DESC(abmlevel,
-		 "ABM level (0 = off, 1-4 = backlight reduction level, -1 auto (default))");
+		 "ABM level (0 = off (default), 1-4 = backlight reduction level, -1 auto)");
 module_param_named(abmlevel, amdgpu_dm_abm_level, int, 0444);
 
 int amdgpu_backlight = -1;
-- 
2.48.1

